pkg/sql/sem/tree/type_check.go

```diff
@@ -3024,11 +3025,45 @@ func getMostSignificantOverload(
 			formatCandidates(expr.Func.String(), overloads, filter),
 		)
 	}
+
 	checkAmbiguity := func(oImpls []uint8) (QualifiedOverload, error) {
 		if len(oImpls) != 1 {
+			// Since there are ambiguity errors when udf with the same family
+			// parameter types, function signature has to be sorted out
+			// to directly match specific udf function for execution, e2e tests added
+			var expTypes []*types.T
+			for _, exp := range typedInputExprs {
+				expTypes = append(expTypes, exp.ResolvedType())
+			}
+
+			udfExactDup := false
+			matchIdx := -1
+			seen := 0
+			for k, idx := range oImpls {
+				candidate := overloads[idx]
+				srcParams := candidate.params()
+				if len(srcParams.Types()) == 0 {
```

Comment:
The srcParam length check was just needed to avoid regression unit test function "TestGetMostSignificantOverload", where prepared overloads have "ParamTypes{}" as the Types field. It seems that in real-life scenario it won't be necessary though.